core/include/scipp/core/element/comparison.h

```diff
@@ -10,28 +10,55 @@
 #include "scipp/common/overloaded.h"
 #include "scipp/core/element/arg_list.h"
 #include "scipp/core/transform_common.h"
+#include "scipp/core/values_and_variances.h"
 
 /// Operators to be used with transform and transform_in_place to implement
 /// operations for Variable.
 namespace scipp::core::element {
 
-constexpr auto is_approx = overloaded{
-    transform_flags::no_out_variance,
-    arg_list<double, float, int64_t, int32_t, std::tuple<double, double, float>,
-             std::tuple<int64_t, int32_t, int32_t>,
-             std::tuple<int64_t, int64_t, int32_t>,
-             std::tuple<int64_t, int32_t, int64_t>,
-             std::tuple<int32_t, int32_t, int64_t>,
-             std::tuple<int32_t, int64_t, int64_t>>,
-    [](const units::Unit &x, const units::Unit &y, const units::Unit &t) {
-      expect::equals(x, y);
-      expect::equals(x, t);
-      return units::dimensionless;
-    },
-    [](const auto &x, const auto &y, const auto &t) {
-      using std::abs;
-      return abs(x - y) <= t;
-    }};
+using is_approx_types_t = arg_list_t<
+    double, float, int64_t, int32_t, std::tuple<double, double, float>,
+    std::tuple<float, float, double>, std::tuple<int64_t, int64_t, double>,
+    std::tuple<int32_t, int32_t, double>, std::tuple<int32_t, int64_t, double>,
+    std::tuple<int64_t, int32_t, double>, std::tuple<int64_t, int32_t, int32_t>,
+    std::tuple<int64_t, int64_t, int32_t>,
+    std::tuple<int64_t, int32_t, int64_t>,
+    std::tuple<int32_t, int32_t, int64_t>,
+    std::tuple<int32_t, int64_t, int64_t>>;
+
+constexpr auto is_approx_units = [](const units::Unit &x, const units::Unit &y,
+                                    const units::Unit &t) {
+  expect::equals(x, y);
+  expect::equals(x, t);
+  return units::dimensionless;
+};
+
+struct is_close_flags_t
+    : public transform_flags::expect_all_or_none_have_variance_t,
+      public transform_flags::no_out_variance_t {
```

Comment:
Do we need the inplace/out version? If not, this flag can be removed?

Reply:
I don't think we need an out variant.